### PR TITLE
3250 Improve example generation

### PR DESCRIPTION
**Pull request by AdamHawtin**
# Motivation and Context

Improve the polish examples script. The tool we're using to generate examples creates lorem ipsum text for every field, even where the schema supplies examples. We then loop over the examples, creating a full event version of each for each example survey type and adding in our own data in a more suitable shape. 

This change refactors that code to hopefully make it a little easier to maintain and extend, and adds a new chunk of code to replace the completely random lorem ipsum with a choice of example values provided in the schema, so that for more basic, non survey type specific examples we can simply put the example data in the schema and have it appear in the example events without having to add further hardcoded examples into the script.

Suggestion for a future enhancement: Avoid regenerating the examples if they still conform to the schema. At the moment every single example JSON is recreated every time the script runs which bloats every change we make, it would be much nicer if it only regenerated for the schemas which had changed, but that will take a chunk more work.

# What has changed

- Refactor python polish examples script
- Add function to use schema examples in example JSON
- Use pipenv for python dependency management
- Update README

# How to test?

Check the branch, install dependencies as directed in the README and run `./regen.sh`, it should regenerate the examples without issue. Review the changes to the python script, it should now look a bit simpler to add more specific examples where we need to.

# Links

https://trello.com/c/qnqkFPV1/3250-find-a-better-way-of-representing-json-schema-examples-1-day

### CODE REVIEW

File: polish_example_json.py [hunk 1]
```diff
@@ -2,148 +2,188 @@
 import random
 import string
 from functools import reduce
+from pathlib import Path
+from typing import Dict
 
+CURRENT_DRAFT_VERSION = '0.6.0-DRAFT'
```
Comment (AdamHawtin) [inline polish_example_json.py:8]: This file is where the main important changes are.

File: polish_example_json.py [hunk 2]
```diff
@@ -2,148 +2,188 @@
...
+
+def to_camel_case(snake_str):
+    components = snake_str.split('-')
+    return reduce(lambda x, y: x + y.title(), components[1:], components[0])
+
 
-                with open(f'examples/{survey_type["type"]}/{event_item["event"]}.example.json', 'w+') as example_file:
-                    json.dump(event, example_file, indent=2)
+if __name__ == '__main__':
+    main()
```
Comment (AdamHawtin) [inline polish_example_json.py:189]: Highlighting these as the main, important changes to review as they are somewhat buried in the PR changes view.

File: polish_example_json.py (hunk 1 above)
Comment (AdamHawtin) [inline polish_example_json.py:8]: Highlighting this as this file contains the important changes, and it's buried in the `Files Changed` view.

**Review by ryangrundy7 — APPROVED** [review]
Looks good to me 👍

**Review by liamtoozer — APPROVED** [review]
Great stuff! 👍

**Pull request merged**